### PR TITLE
Fixes #31951 - f-m online backup to skip /var/lib/pulp/cache

### PR DESCRIPTION
**Pull request by pmoravec**
Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

### CODE REVIEW

**Comment by upadhyeammit** [comment]
Thanks for patience!
Could you move both exclude directories in pulp2 feature and call it in the procedure? something like below,

In file : foreman_maintain/definitions/features/pulp2.rb
~~~
def exclude_for_backup
  ['var/lib/pulp/katello-export', 'var/lib/pulp/cache']
end
~~~

In file: definitions/procedures/backup/pulp.rb
~~~
:exclude => feature(:pulp2).exclude_for_backup,
~~~

**Review by upadhyeammit — CHANGES_REQUESTED** [review]
Added small change, request to check!
thank you!

**Comment by pmoravec** [comment]
Thanks for the feedback, good point the change is pulp-2 related only. PR updated.

File: definitions/features/pulp2.rb [hunk 1]
```diff
@@ -34,4 +34,8 @@ def config_files
       '/etc/default/pulp_workers'
     ]
   end
+
+  def exclude_for_backup
+    ['var/lib/pulp/katello-export', 'var/lib/pulp/cache']
```
Comment (upadhyeammit) [inline definitions/features/pulp2.rb:39]: The tar command is being run from `/var/lib/pulp` and hence exclude never worked. Request to change method like below,
~~~
def exclude_from_backup
  # Exclude /var/lib/pulp/katello-export and /var/lib/pulp/cache
  %w[katello-export cache]
end
~~~

You will also need to change `feature(:pulp2).exclude_for_backup` to `feature(:pulp2).exclude_from_backup`

**Review by upadhyeammit — CHANGES_REQUESTED** [review]

**Comment by pmoravec** [comment]
Sorry for the typos :)

**Review by upadhyeammit — APPROVED** [review]

**Pull request merged**